documentation/customisation.md

```diff
@@ -0,0 +1,139 @@
+
+Please note that certain JARS will go into certain paths. You shouldn't add a Oracle JAR to the `ext/` folder, for example. If you are unsure where a JAR should be added, please get in touch with our support team. 
+
+NOTE: `${APP_HOME}` points to the path `/opt/xebialabs/xl-deploy-server` by default
+
+```docker
+FROM xebialabs/xl-deploy:9.5.0
+
+###################################################################################
+# PLUGINS                                                                         #
+# Plugins should be placed under ${APP_HOME}/default-plugins/ #
+###################################################################################
+
+COPY --chown=10001:0 files/xld-liquibase-plugin-5.0.1.xldp /opt/xebialabs/xl-deploy-server/default-plugins/
+
+# Add plugin from url
+ADD --chown=10001:0 https://dist.xebialabs.com/public/community/xl-deploy/command2-plugin/3.9.1-1/command2-plugin-3.9.1-1.jar /opt/xebialabs/xl-deploy-server/default-plugins/
+
+##########################################################################
+# EXTENSIONS                                                             #
+# Extensions should be placed under ${APP_HOME}/ext
+##########################################################################
+ADD --chown=10001:0 files/ext /opt/xebialabs/xl-deploy-server/ext/
+
+##########################################################################
+# HOTFIXES                                                               #
+##########################################################################
+ADD --chown=10001:0 files/lib-hotfix.jar /opt/xebialabs/xl-deploy-server/hotfix/lib/
+ADD --chown=10001:0 files/plugin-hotfix.jar /opt/xebialabs/xl-deploy-server/hotfix/plugin/
+ADD --chown=10001:0 files/sattelite-lib-hotfix.jar /opt/xebialabs/xl-deploy-server/hotfix/sattelite-lib/
+
+##########################################################################
+# LIBRARIES                                                              #
+##########################################################################
+ADD --chown=10001:0 files/ojdbc6.jar /opt/xebialabs/xl-deploy-server/lib/
+```
+
+Please note that there are seperate hotfix directories where you can place hotfix JARs, depending on the type of hotfix. Please consult our support team in case you are uncertain. 
+
+For an overview of how `ADD` and `COPY` works, please refer to [the documentation](https://docs.docker.com/engine/reference/builder/#add)
+
+Once you are satisfied with your Dockerfile, run the following command in the same directory
+
+`docker build -t xl-deploy-custom:9.5.0 .`
+
+Always use [semver](https://semver.org/) to version your docker images. This is to ensure future compatability with one of our other tools, `xl up`.
+
+This command will build and tag a docker image for you. If you would like to run this image locally, you can do this now by running
+
+`docker run -it --rm -p 4601:4601 -e ACCEPT_EULA=Y xl-deploy-custom:9.5.0`
```

Comment:
- This will not work as XLDeploy uses port 4516 not 4601
- Also based on above comment custom image tag should be 9.0
- i think it is better to add Admin password here using -e "ADMIN_PASSWORD=desired_admin_password"

Reply:
indeed:
 * This will not work as XLDeploy uses port 4516 not 4601
* i think it is better to add Admin password here using -e "ADMIN_PASSWORD=desired_admin_password"

should be fixed